Gems/ROS2/Code/Include/ROS2/Sensor/ROS2SensorComponent.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "ROS2/ROS2GemUtilities.h"
+#include "SensorConfiguration.h"
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+
+namespace ROS2
+{
+    //! Captures common behavior of ROS2 sensor Components.
+    //! Sensors acquire data from the simulation engine and publish it to ROS2 ecosystem.
+    //! Derive this Component to implement a new ROS2 sensor. Each sensor Component requires ROS2FrameComponent.
+    class ROS2SensorComponent
+        : public AZ::Component
+        , public AZ::TickBus::Handler // TODO - high resolution tick source?
```

Comment:
This TODO seems like a question and not an actual TODO.

The TickBus ticks with microsecond resolution using the steady_clock via the ITimeSystem in [ComponentApplication::Tick](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp#L1472-L1475).

This TODO should be able to be removed then.